tests/templates/test_embeddings.py

```diff
@@ -421,21 +421,24 @@ def circuit(x=None):
         res = circuit(x=features[:n_subsystems])
         assert np.allclose(res, target, atol=tol, rtol=0)
 
-    @pytest.mark.parametrize('features, weights, target', [([np.pi/2, 0], [[np.pi, 0, 0]], [-1, 1]),
-                                                           ([0, np.pi/2], [[np.pi, 0, 0]], [1, -1]),
-                                                           ([np.pi/2, np.pi/2], [[np.pi, 0, 0]], [-1, -1])])
-    def test_qaoa_embedding_state_features_and_zz(self, features, weights, target, tol):
-        """Checks the state produced by QAOAEmbedding() is correct if the features and weights are nonzero.
+    @pytest.mark.parametrize('weights, target', [([[np.pi, 0, 0], [0, 0, 0]], [1, 1]),
+                                                           ([[np.pi/2, 0, 0], [0, 0, 0]], [0, 0]),
+                                                           ([[0, 0, 0], [0, 0, 0]], [-1, -1])])
+    def test_qaoa_embedding_state_features_and_zz(self, weights, target, tol):
+        """Checks the output if the features and entangler weights are nonzero.
         Uses RY local fields."""
 
         dev = qml.device('default.qubit', wires=2)
 
         @qml.qnode(dev)
         def circuit(x=None):
-            QAOAEmbedding(features=x, weights=weights, wires=range(2), local_field='Y')
+            QAOAEmbedding(features=x, weights=weights, wires=range(2))
+            # Undo the final X block
+            qml.RX(-np.pi/2, wires=0)
+            qml.RX(-np.pi/2, wires=1)
```

Comment:
How come these `RX` operations were needed?

Reply:
That took me a while to remember, too, but the template is programmed to have a final block of Hadamards after the last layer - but those were undoing the dependency of the output on the ZZ gate.

Reply:
Hmm, would it not be the `RX` operations from the feature encoding part? :thinking: (as for 2 wires with 2 features we'd have no remaining wires for `Hadamards`)
https://github.com/XanaduAI/pennylane/blob/349221782c932efcba76d512c817e86596c87215/pennylane/templates/embeddings/qaoa.py#L53

Btw. checked the analytic value and it matches well! :100:

Reply:
Oh, yes, by Hadamards I meant the ``RX`` operations with angles ``np.pi/2``...